File: minyoung/BOJ_18258.java
import java.util.StringTokenizer;
import java.util.Queue;
import java.util.LinkedList;
import java.io.BufferedReader;
import java.io.InputStreamReader;


public class Main
{
	public static int n;
	
	public static void main (String[] args) throws java.lang.Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		StringBuilder sb = new StringBuilder();
		
		Queue <Integer> queue = new LinkedList<>();
		
		
		n = Integer.parseInt(br.readLine());
			
		int last = 0;
		
		for(int i=0;i<n;i++) {
			st = new StringTokenizer(br.readLine());
			String str = st.nextToken();
			switch(str) {
				case "push":
					last = Integer.parseInt(st.nextToken());
					queue.offer(last);
					break;
				case "pop":
					if(!queue.isEmpty())
						sb.append(queue.poll());
					else
						sb.append("-1");
					sb.append("\n");
					break;
				case "size":
					sb.append(queue.size());
					sb.append("\n");
					break;
				case "empty":
					if(queue.isEmpty())
						sb.append(1);
					else
						sb.append(0);
					sb.append("\n");
					break;
				case "front":
					if(!queue.isEmpty())
						sb.append(queue.peek());
					else
						sb.append("-1");
					sb.append("\n");
					break;
				case "back":
					if(!queue.isEmpty())
						sb.append(last);
					else
						sb.append("-1");
					sb.append("\n");
					break;
					
			}

		}
		
	
		System.out.println(sb);
	}
}

File: hyerang0125/0x06/boj_18258.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    cin.tie(0); cin.sync_with_stdio(0);

    int x, n;
    string str;
    queue<int> q;

    cin >> n;
    for (int i = 0; i < n; i++)
    {
        cin >> str;
        if (str == "push")
        {
            cin >> x;
            q.push(x);
        }
        else if (str == "pop")
        {
            if (q.empty())
                cout << "-1\n";
            else
            {
                cout << q.front() << "\n";
                q.pop();
            }
        }
        else if (str == "size")
        {
            cout << q.size() << "\n";
        }
        else if (str == "empty")
        {
            if (q.empty())
                cout << "1\n";
            else
                cout << "0\n";
        }
        else if (str == "front")
        {
            if (q.empty())
                cout << "-1\n";
            else
                cout << q.front() << "\n";
        }
        else if (str == "back")
        {
            if (q.empty())
                cout << "-1\n";
            else
                cout << q.back() << "\n";
        }
    }

    return 0;
}